### PR TITLE
vulkan: update to 1.2.178

### PR DESCRIPTION
**Pull request by neheb**

### CODE REVIEW

File: mingw-w64-vulkan-loader/002-proper-def-files-for-32bit.patch [hunk 1]
```diff
@@ -60,19 +60,6 @@
          target_link_libraries(vulkan OneCoreUAP.lib LIBCMT.LIB LIBCMTD.LIB LIBVCRUNTIME.LIB LIBUCRT.LIB)
          set_target_properties(vulkan PROPERTIES LINK_FLAGS "/NODEFAULTLIB")
      else()
---- Vulkan-Loader-1.2.158.orig/loader/loader.h
```
Comment (lazka) [inline mingw-w64-vulkan-loader/002-proper-def-files-for-32bit.patch:63]: what's this about?

Comment (neheb) [reply mingw-w64-vulkan-loader/002-proper-def-files-for-32bit.patch:63]: Breaks compilation with clang as clang expects dllexport in the declaration only.

Comment (lazka) [reply mingw-w64-vulkan-loader/002-proper-def-files-for-32bit.patch:63]: this breaks the ABI for 32bit I think, at least the symbols are different here

**Comment by neheb** [comment]
bleh too many.

**Comment by Biswa96** [comment]
I have fixed the issue. I will try to discuss this with upstream. The whole patch is not wrong. The trick is to define `__declspec(dllexport)` at the time of building DLLs _only_. clang is right here. Thanks @mati865 for the hint 😍

File: mingw-w64-vulkan-loader/002-proper-def-files-for-32bit.patch (hunk 1 above)
Comment (Biswa96) [reply mingw-w64-vulkan-loader/002-proper-def-files-for-32bit.patch:63]: The patch file name is somewhat misleading. It does not do anything with 32bit building. Will try upstreaming.

**Pull request closed without merging**